src/common/lib/file-system/s3-file-system.ts

```diff
@@ -0,0 +1,115 @@
+import {S3Client, GetObjectCommand, PutObjectCommand, DeleteObjectCommand, HeadObjectCommand, type S3ClientConfig} from "@aws-sdk/client-s3";
+
+import type {Config} from "../../../config/config";
+import type {ILoggerService} from "../../../service/logger/i-logger-service";
+import {RealFileSystem} from "./real-file-system.js";
+import {fallbackSuper} from "../decorator/simple-s3-switcher.js";
+import {error2false} from "../decorator/error2false.js";
+
+export class S3FileSystem extends RealFileSystem {
+	private s3Client: S3Client;
+	private bucket: string;
+
+	constructor(private readonly config: Config, private readonly logger: ILoggerService) {
+		super();
+
+		if (!config.s3Storage?.bucket) {
+			throw new Error("S3_BUCKET must be configured when using S3 storage");
+		}
+
+		this.bucket = config.s3Storage.bucket;
+
+		const s3Config: S3ClientConfig = {
+			region: config.s3Storage.region || "us-east-1"
+		};
+
+		// If access key is provided, use it
+		if (config.s3Storage.accessKeyId && config.s3Storage.secretAccessKey) {
+			s3Config.credentials = {
+				accessKeyId: config.s3Storage.accessKeyId,
+				secretAccessKey: config.s3Storage.secretAccessKey
+			};
+		}
+
+		// If a custom endpoint is provided (e.g. MinIO, R2, etc.)
+		if (config.s3Storage.endpoint) {
+			s3Config.endpoint = config.s3Storage.endpoint;
+			s3Config.forcePathStyle = config.s3Storage.forcePathStyle;
+		}
+
+		this.s3Client = new S3Client(s3Config);
+	}
+
+	private getS3Key(path: string): string {
+		// Remove the leading / and s3:// prefix
+		return path.replace(/^(s3:\/\/|\/?)/, "").replace(/^\/+/, "");
+	}
+
+	public isS3Path(path: string): boolean {
+		return path.startsWith("s3://");
+	}
+
+	public isValidPath(path: string): boolean {
+		return this.isS3Path(path);
+	}
+
+	@fallbackSuper
+	@error2false
+	async exists(path: string): Promise<boolean> {
```

Comment:
```suggestion
	@error2false
	@fallbackSuper
	async exists(path: string): Promise<boolean> {
```
装饰器对目标的包裹是由外到内，但最终逻辑执行时是由内到外的。因此，装饰器逻辑越接近函数参数的放在内层，越接近返回值或异常的放在外层。